linux/vhba_hot_add_remove/freebsd_create_disk_partition.yml

```diff
@@ -30,14 +30,105 @@
+# The disks maybe exchanged after reboot
+- name: "Set boot disk for FreeBSD 32 bit"
+  when:
+    - guest_os_bit == "32-bit"
+    - add_new_controller is defined and add_new_controller
+  block:
+    - name: "Initialize the fact of device list before reboot and after reboot"
+      ansible.builtin.set_fact:
+        guest_device_list_before_reboot: []
+        guest_device_list_after_reboot: []
+
+    - name: "Get block device list before reboot"
+      ansible.builtin.shell: "/tmp/lsblk.sh -d | grep -v SIZE | grep -v cd0"
+      delegate_to: "{{ vm_guest_ip }}"
+      changed_when: false
+      register: lsblk_result
+
+    - name: "Set the fact of block device list before reboot"
+      ansible.builtin.set_fact:
+        guest_device_list_before_reboot: >
+          {{  
+              guest_device_list_before_reboot | union([
+                dict(['name', 'size', 'model'] |
+                      zip_longest(item.split(None, maxsplit=2), fillvalue=''))
+              ])
+          }}
+      with_items: "{{ lsblk_result.stdout_lines }}"
+
+    - name: "Boot order may be changed after hot-add disk to new controller for OS with BIOS firmware"
+      ansible.builtin.debug:
+        msg:
+          - "The disk boot order may be changed after hot-add new disk which causes failure of powering on VM."
+          - "Specify the boot disk as workaround. "
+
+    - include_tasks: ../../common/vm_set_boot_options.yml
+      vars:
+        boot_hdd_name: "Hard disk 1"
+        boot_order_list:
+          - disk
+
+    - name: "Reboot guest OS to check file content again"
+      include_tasks: ../utils/reboot.yml
+
+    - name: "Get block device list after reboot"
+      ansible.builtin.shell: "/tmp/lsblk.sh -d | grep -v SIZE | grep -v cd0"
+      delegate_to: "{{ vm_guest_ip }}"
+      changed_when: false
+      register: lsblk_result
+
+    - name: "Set the fact of block device list after reboot"
+      ansible.builtin.set_fact:
+        guest_device_list_after_reboot: >
+          {{  
+              guest_device_list_after_reboot | union([
+                dict(['name', 'size', 'model'] |
+                      zip_longest(item.split(None, maxsplit=2), fillvalue=''))
+              ])
+          }}
+      with_items: "{{ lsblk_result.stdout_lines }}"
+
+    - name: "Set fact of changed disk list after reboot"
+      ansible.builtin.set_fact:
+        guest_disk_changed_list: "{{ guest_device_list_before_reboot | difference(guest_device_list_after_reboot) }}"
+
+    - name: "Exchange disk name between boot disk and new disk if it's changed after reboot"
+      when: guest_disk_changed_list | length == 2
+      block:
+        - name: "Set fact of new disk info after it's changed"
+          ansible.builtin.set_fact:
+            new_guest_disk_info: >
+              {%- if item.size == '1.0G' -%}{{ item }}
+              {%- else -%}{{ new_guest_disk_info }}
+              {%- endif -%}
+          with_list: "{{ guest_device_list_after_reboot }}"
```

Comment:
FreeBSD devices should also have UUID for each one. You can check whether there is a way to get new disk's UUID and find it by UUID after reboot.

Reply:
Fixed in pull request 491.